core/pom.xml

```diff
@@ -75,6 +77,15 @@
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+    </dependency>
+    <dependency>
```

Comment:
1. True, I added it to write my manual test class, but it's not used elsewhere and might not even be used in future tests. Removing it.
2. Yes. I don't see any need to have it out of the box because a) Adding a JMX reporter is trivial (two lines of code) and b) it's not recommended in production so users are likely to use something else.
3. I don't understand, as far as I can tell the JMX reporter does already create a hierarchy based on the dot separators in the metric names. When I look up the MBeans in jvisualvm, `retries` is a child of `127_0_0_1_9042`, etc.

Reply:
> as far as I can tell the JMX reporter does already create a hierarchy based on the dot separators in the metric names

Was this available since metrics 3? I didn't know. OK then.

Reply:
I double-checked and actually you're right, I stand corrected.

But there is a way to do it from outside the driver: build the `JmxReporter` with an `ObjectNameFactory` that turns every dot-separated component into a subdirectory:

```java
ObjectNameFactory objectNameFactory = (type, domain, name) -> {
  StringBuilder objectName = new StringBuilder(domain).append(':');
  List<String> nameParts = Splitter.on('.').splitToList(name);
  int i = 0;
  for (String namePart : nameParts) {
    boolean isLast = (i == nameParts.size() - 1);
    String key =
        isLast ? "name" : Strings.padStart(Integer.toString(i), 2, '0');
    objectName.append(key).append('=').append(namePart);
    if (!isLast) {
      objectName.append(',');
    }
    i += 1;
  }
  try {
    return new ObjectName(objectName.toString());
  } catch (MalformedObjectNameException e) {
    throw new RuntimeException(e);
  }
};

JmxReporter reporter =
    JmxReporter.forRegistry(session.getMetricRegistry())
        .inDomain("com.datastax.oss.driver")
        .createsObjectNamesWith(objectNameFactory)
        .build()
```

Reply:
Yes that's basically how we configured the JMX reporter in DSBulk. 👍